x/precisebank/module.go

```diff
@@ -0,0 +1,165 @@
+package precisebank
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/evm/x/precisebank/client/cli"
+	"github.com/cosmos/evm/x/precisebank/keeper"
+	"github.com/cosmos/evm/x/precisebank/types"
+
+	"cosmossdk.io/core/appmodule"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+// ConsensusVersion defines the current module consensus version.
+const ConsensusVersion = 1
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+	_ module.HasABCIGenesis = AppModule{}
+
+	_ appmodule.AppModule   = AppModule{}
+	_ module.HasABCIGenesis = AppModule{}
```

Comment:
An invariant has been added to this module. Would it make sense to also add `HasInvariants`? [ref. bank](https://github.com/cosmos/cosmos-sdk/blob/98164408932ee74369b7ce18aa13bb1a1a6ba76b/x/bank/module.go#L44)